### PR TITLE
Scryfall downloader

### PR DESCRIPTION
**Pull request by DorkSoul**
# Description

Had a program I made to download cards from scryfall for myself and decided to see about adding it to mpc autofill for everyone to use.

No idea if this is something that would be wanted here but was an interesting project and something I will use myself with your fantastic program

This is an addition to allow the user to use a decklist exported from sites like moxfield and saved in a .txt file.

If this is something that is useful to you ill work on doing some actually tests for coverage rather then jsut manual testing

Didn't see a readme that I would be updating just a link to the github wiki which I cant seem to add to in this. Let me know where to add the readme for the two supported .txt format and any other info.

# Checklist

- [x] I have installed `pre-commit` and installed the hooks with `pre-commit install` before creating any commits.
- [ ] I have updated any related tests for code I modified or added new tests where appropriate.
- [x] I have manually tested my changes as follows:
  - tested on multiple normal decklists using both XML and TXT.
  - tested on a decklist with one of each type of card layout from the scryfall api.
  - tested on a decklist with one of each type of card layout from the MPC autofill.
  - tested on a decklist with multiple meld cards. some with both some with one.
- [ ] I have updated any relevant documentation or created new documentation where appropriate.

### CODE REVIEW

**Comment by ndepaola** [comment]
Thank you for submitting this PR! I'll aim to review this over the next couple of weeks.

**Comment by DorkSoul** [comment]
added a load of tests and fixed a border issue on the back of scryfall meld cards. Should have most of the coverage now. Still a few places I could probably add a test for edge cases or failures but for now that's most of them.

**Comment by ndepaola** [comment]
Hey mate, sorry for the quietness on this. Life has been pretty hectic for me lately so I haven't had much time to devote to open source dev. I do still plan on reviewing this code so we can get it in but can't really give you a timeframe on when that will be 😭

File: desktop-tool/src/constants.py [hunk 1]
```diff
@@ -181,3 +181,8 @@ class TargetSites(Enum):
 THREADS = 5  # shared between CardImageCollections
 
 POST_LAUNCH_HTML_FILENAME = "post-launch.html"
+
+# Card physical dimensions for border calculation
+CARD_WIDTH_INCHES = 2.5
```
Comment (alex-taxiera) [inline desktop-tool/src/constants.py]:
```suggestion
CARD_WIDTH_INCHES = 2.48
CARD_HEIGHT_INCHES = 3.46
BORDER_INCHES = 0.12
```
these are the actual values that MPC uses (or at least a bit closer)
